packages/core/src/common/props.ts

```diff
@@ -23,6 +23,13 @@ export type HTMLDivProps = React.HTMLAttributes<HTMLDivElement>;
  */
 export type HTMLInputProps = React.InputHTMLAttributes<HTMLInputElement>;
 
+/**
+ * Alias for a `JSX.Element` or a value that renders nothing.
+ *
+ * In React, `boolean`, `null`, and `undefined` do not produce any output.
+ */
+export type OptionalElement = JSX.Element | boolean | null | undefined;
```

Comment:
how about just going straight to `React.ReactNode`? seems like that's what you support in the implementation of icon

Reply:
crucially no, we do not support fragments or arrays, nor do i think we should in this case. I really am going for `Element | nothing`.

Reply:
i just don't see a use case for passing multiple icons, from a UX perspective.